gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+        for p in processes:
+            p.terminate()
+
+    def generate_topic_models(self, num_models, random_states=None, pipe=None):
+        """Will train the topic models, that form the ensemble.
+
+        Parameters
+        ----------
+        num_models : number
+            number of models to be generated
+        random_states : list
+            list of numbers or np.random.RandomState objects.
+            Will be autogenerated based on the ensembles RandomState
+            if None (default).
+        pipe : multiprocessing.pipe
+            Default None. If provided, will send the trained models over
+            this pipe. If memory friendly, it will only send the ttda.
+
+        """
+
+        if pipe is not None:
+            logger.info("Spawned worker to generate {} topic models".format(num_models))
+
+        if random_states is None:
+            random_states = [self.random_state.randint(MAX_RANDOM_STATE) for _ in range(num_models)]
+
+        assert len(random_states) == num_models
+
+        kwArgs = self.gensim_kw_args.copy()
+
+        tm = None  # remember one of the topic models from the following
+        # loop, in order to collect some properties from it afterwards.
+
+        for i in range(num_models):
+
+            kwArgs["random_state"] = random_states[i]
+
+            tm = self.topic_model_kind(**kwArgs)
+
+            # adds the lambda (that is the unnormalized get_topics) to ttda, which is
+            # a list of all those lambdas
+            self.ttda = np.concatenate([self.ttda, tm.get_topics()])
+
+            # only saves the model if it is not "memory friendly"
+            if not self.memory_friendly_ttda:
+                self.tms += [tm]
+
+        # use one of the tms to get some info that will be needed later
+        self.sstats_sum = tm.state.sstats.sum()
+        self.eta = tm.eta
+
+        if pipe is not None:
+            # send the ttda that is in the child/workers version of the memory into the pipe
+            # available, after generate_topic_models has been called in the worker
+            if self.memory_friendly_ttda:
+                # remember that this code is inside the worker processes memory,
+                # so self.ttda is the ttda of only a chunk of models
+                pipe.send(self.ttda)
+            else:
+                pipe.send(self.tms)
+
+            pipe.close()
+
+    def asymmetric_distance_matrix_worker(self, worker_id, ttdas_sent, n_ttdas, pipe, threshold, method):
+        """ worker, that computes the distance to all other nodes
+        from a chunk of nodes. https://stackoverflow.com/a/1743350
+        """
+
+        logger.info("Spawned worker to generate {} rows of the asymmetric distance matrix".format(n_ttdas))
+        # the chunk of ttda that's going to be calculated:
+        ttda1 = self.ttda[ttdas_sent:ttdas_sent + n_ttdas]
+        distance_chunk = self.calculate_asymmetric_distance_matrix_chunk(ttda1=ttda1, ttda2=self.ttda,
```

Comment:
This indentation is a bit ugly. Please do something like:

```python
kwargs = dict(ttda1=ttda1, ttda2=...)
distance_chunk = self.calculate_asymmetric_distance_matrix_chunk(**kwargs)
```

Also, this is a sign that your method lines are perhaps a little long. Consider shortening them.

Finally, think about scoping your methods. Are these methods public or _internal?

Reply:
done